### ISSUE TITLE
Not interacting with matrix

### ISSUE THREAD

**Issue by demipixel** [zeitungen/node-rpi-rgb-led-matrix#14]
I'm using node v7.8.0 and the matrix doesn't appear to be lighting up no matter what I use. I've confirmed the size (32x32) and it's clear it's at least trying because it fails without sudo (i.e. requiring permissions, isn't just doing nothing).

I've essentially just been trying this:

```
> const LedMatrix = require('node-rpi-rgb-led-matrix')
undefined
> const matrix = new LedMatrix(32);
undefined
> matrix.getWidth()
32
> matrix.getHeight()
32
> matrix.fill(255, 50, 100);
undefined
> matrix.setPixel(0, 0, 255, 255, 255)
undefined
```

The python API works. The demos don't work if I directly go through node_module's into this library's rpi-rgb-led-matrix. Copying/linking to an existing (working) one makes the demos work at that location, but the node library still doesn't function.

No errors, hence why I'm confuzzled.

**Comment by zeitungen** [comment]
Unfortunately, I have only tested the lib with nodejs 0.10 ..
So, maybe, this lib doesn't work for now with recent nodejs version.

I hope I can manage in this few days some time to look into that problem ...

To reproduce this issue, can you tell me how you have install the lib into your project ? (via npm ? via git and github ?)

**Comment by demipixel** [comment]
I'm using npm.

I'll try using an older version of node when I get the chance.

**Comment by michaelwschultz** [comment]
@demipixel did you ever have any luck with running an older version of Node?

I'm running 4.2.1 and am having similar issues. When I run as sudo, index runs but doesn't manipulate the board. When I run as a user I get /dev/mem Permission Denied.

**Comment by hzeller** [comment]
Note, if you have a newer version of Raspbian, the **old** rpi-rgb-led-matrix library this node library is linked to will *not* work anymore.

I really strongly suggest to update this node library to use the latest version of rpi-rgb-led-matrix @zeitungen .

In the meantime: for newer Raspbian's to work, you need to change the `==` to a `>=` in the `(mem_size == 0x3F000000)` line in IsRaspberryPi() in lib/gpio.cc.

The Node library is linking a very old version of rpi-rgb-led-matrix; the line to change in that old version is [here in lib/gpio:136](https://github.com/hzeller/rpi-rgb-led-matrix/blob/5cdf1b4954aed4ea8b9d7fb57f3d1f410201408d/lib/gpio.cc#L136).

**Comment by michaelwschultz** [comment]
@hzeller Thanks for the tip. No change in my error.  Still seeing...

`/bin/sh: 1: nodejs: not found
gyp: Call to 'nodejs -e "require('nan')"' returned exit status 127 while in binding.gyp. while trying to load binding.gyp`

**Comment by hzeller** [comment]
Yeah, I can't help with Node.js stuff - I don't know Javascript or the Node environment.
The patch above just makes sure that the old matrix library will work on a new Raspbian.

**Comment by michaelwschultz** [comment]
@hzeller no worries. Really appreciate you writing back so quickly!

**Comment by hzeller** [comment]
Same timezone helps I guess @michaelwschultz :)
